### DISCUSSION TITLE
List of all attributes?

### DISCUSSION THREAD

**Discussion by renyuneyun** [TriliumNext/Trilium#1988] [Q&A] [answered]
Sometimes I use the attributes (labels, more precisely) as tags for notes.
However, I don't see anywhere to list **all** existing attributes so that I can filter on them.
Because this is a common feature of related applications, I think it's most likely that I failed to find it.

Can anyone give any suggestions?

**Comment by zadam** [comment, marked as answer]
Hi, this is not really "core" functionality, but the demo document contains an example script listing all existing attributes:

![image](https://user-images.githubusercontent.com/617641/120229127-c991a300-c24c-11eb-8823-6b314afd539d.png)

**Comment by bert010** [comment]
(edit: you can also do this in a normal search with ""tag = " as search query but then the notes are not ordered by tag)

as workaround I use the collection views plugin to generate a list of all my notes labeled tag. (eg #tag=cmd #tag = idee etc)

a note with these attributes;
#view=table ~renderNote=Collection Views #query="#tag = " #groupBy=tag #attribute=tag #sort=tag  

results in a list like this:

![image](https://user-images.githubusercontent.com/84742924/120229951-95b77d00-c24e-11eb-863a-e70e8c39155d.png)

**Reply by ghost** [reply]
I have tried creating a render note with those attributes with the collection views plugin but it is not working. It gives me this error: Invalid attribute "Views" in "...nderNote=Collection Views #query="#tag = " #..."

You say this: as workaround I use the collection views plugin to generate a list of all my notes labeled tag. (eg #tag=cmd #tag = idee etc). Could you make a screenshot of that. I think I am doing something wrong. Thanks.

**Reply by bert010** [reply]
it's exactly this line I use :  #sort=tag #view=table #attribute=tag #groupBy=tag ~renderNote=
Collection Views  #query="#tag = " 

in your line I see  #query="#tag = " #..."  at the end, maybe thats the culprit?

**Reply by mabeyj** [reply]
The error is coming from trying to parse the `~renderNote=Collection Views` text, so you'll have to create that attribute using the "+" button and then selecting the Collection Views note. Copying and pasting that text doesn't create the relation correctly.

![image](https://user-images.githubusercontent.com/773401/125712457-4152f705-0650-436c-9ef8-b487438d4cf7.png) 
![image](https://user-images.githubusercontent.com/773401/125713172-5096321d-b8dc-467f-adc0-22694fad9f78.png)

**Reply by bert010** [reply]
thnkz for correcting that!
